pkg/gardenlet/controller/seed/seed.go

```diff
@@ -224,54 +226,76 @@ func (c *Controller) Run(ctx context.Context, workers int) {
 func (c *Controller) startHealthManagement(ctx context.Context) {
 	var (
 		seedName              = confighelper.SeedNameFromSeedConfig(c.config.SeedConfig)
-		seedLabelSelector     labels.Selector
+		listOption            client.ListOption
 		expectedHealthReports int
 		err                   error
 	)
 
+	gardenClient, err := c.clientMap.GetClient(ctx, keys.ForGarden())
+	if err != nil {
+		panic(fmt.Errorf("failed to start health management. Cannot create garden client: %v", err))
+	}
+
 	if seedName != "" {
 		expectedHealthReports = 1
 	} else {
-		seedLabelSelector, err = metav1.LabelSelectorAsSelector(c.config.SeedSelector)
+		seedLabelSelector, err := metav1.LabelSelectorAsSelector(c.config.SeedSelector)
 		if err != nil {
 			panic(err)
 		}
+		listOption = &client.ListOptions{
+			LabelSelector: seedLabelSelector,
+		}
 	}
 
+	logger.Logger.Info("Starting seed health management")
+
 	for {
 		select {
 		case <-ctx.Done():
 			return
-		default:
-			time.Sleep(LeaseResyncGracePeriodSeconds / 2 * time.Second)
-
-			health := true
-
-			if seedName == "" {
-				seedList, err := c.k8sGardenCoreInformers.Core().V1beta1().Seeds().Lister().List(seedLabelSelector)
+		case <-time.After(LeaseResyncGracePeriodSeconds / 2 * time.Second):
+
+			isHealthy := true
+
+			if len(seedName) > 0 {
+				seed := &gardencorev1beta1.Seed{}
+				if err := gardenClient.DirectClient().Get(ctx, kutil.Key(seedName), seed); err != nil {
```

Comment:
Maybe I missed this earlier, but can we please continue to use the cache to avoid so many "real" calls?

Reply:
I just wrote a comment, too fast :)